flake.nix

```diff
@@ -16,7 +16,133 @@
         "x86_64-darwin" = [ "darwin" "x64" ];
         "aarch64-darwin" = [ "darwin" "arm64" ];
       };
+
     in
+    {
+      nixosModules.default = { config, ... }: with nixpkgs; with lib;
+      {
+        options = {
+          services.polykey = {
+            enable = mkEnableOption "Enable the Polykey agent. Users with the `polykey` group or root permissions will be able to manage the agent.";
+
+            recovery = mkEnableOption "Enable recovery mode for the node. Will try to source from a file under the configuration directory called `recovery-code`.";
+
```

Comment:
This still feels like a jank way to enable use of the recovery code. Did you explore any alternatives?

Reply:
There aren't many alternatives here, as we cannot directly pass in parameters when rebuilding (flakes are self-contained). I need to explore more options, I'll write up any I find here that would be a suitable alternative.

Reply:
I have an idea, it involves the `systemd-ask-password` command. When the user runs this command, it will ask for a password to use as a credential, and will store it as a credential Systemd is able to read from in service units. We could then copy that password to a more permanent directory to maintain persistence.

Reply:
Hm I think though my previous idea would pertain more to sourcing a password into it, let me rethink this. Maybe we can output to an environment variable that could be read by the user? Or we just automatically output it if it is generated, and then alert the user that it was generated, and they handle it themselves?

Reply:
But there's also the question of telling the agent whether to actually initiate recovery. I suppose if the user sets an environment variable such as `POLYKEY_RECOVERY` it will recover using the configured file path in the module?

Reply:
Maybe a simple script in the unit file that will load a recovery code from a file to an env variable once and delete that file?

Reply:
I'm not too sure about this, already we read from a file. If it were put into an environment variable, we would need to make that password persistent anyway, and under a user-space I feel like it would just make more sense to let the user handle the file themselves, instead of making a file under some sort of system directory that does not have any read access. Maybe in the case of the system service we can do something like what you suggested, but user-level probably not.

Reply:
Okay, I suppose we can make it so that if we are recovering, we check a directory to see if a recovery code file is present; if it is, we recover using that and delete the file that was provided. Let me implement this.

Reply:
I have made this logic now:
```nix
            recoveryCodePath = "${config.services.polykey.configPath}/recovery-code";
            recoveryCodeExists = builtins.pathExists path;
```

Basically, if the user provides the file in a set location (in this case in the configuration directory), it will automatically determine if a recovery is needed or not. So if the file exists, we recover, and if it doesn't, we create a new vault altogether. Still not sure though about deleting the recovery code file, especially in the case where the user doesn't provide one (the file will be put in said configuration directory and the user can deal with it).

Reply:
What do you mean we create a new vault altogether? What does vaults have to do with recovery codes?

Reply:
Resolved in latest commit, it will now only recover if the user sets the recovery code property themselves.